### ISSUE TITLE
Click/Hover can't scroll to element and fails to click/Hover due to element visibility

### ISSUE THREAD

**Issue by KristianPP** [DevExpress/testcafe#2900]
### Are you requesting a feature or reporting a bug?

Bug

### What is the current behavior?

When attempting to click on an element
```
if(await hp.carouselItem.exists && hp.carouselItem.visible){
    	   await t
    	   .hover(hp.carouselItem)
    	   .click(hp.carouselItem);
    }
```
The page constantly scrolls to the element then scrolls to the very top of the page. In the end due to timeout we get the  element not visible error. This happens both on click and hover.

```

  1) The element that matches the specified selector is not visible.

     Browser: Chrome 68.0.3440 / Windows 7.0.0

        15 |    await t.maximizeWindow()
        16 |    .expect(hp.carousel.exists).ok()
        17 |    .click(hp.consentCloseButton)
        18 |    if(await hp.carouselItem.exists &&
     hp.carouselItem.visible){
        19 |              await t
      > 20 |              .hover(hp.carouselItem)
        21 |              .click(hp.carouselItem);
        22 |    }
        23 |
        24 |});
        25 |
```

### What is the expected behavior?

Element is scrolled to and clicked.

### How would you reproduce the current behavior (if this is a bug)?

One site where i can always reproduce it is adidas uk

### test js

```
import { Selector } from 'testcafe';
import HomePage from '../pages/homePage';
const hp = new HomePage();

fixture `Home Page test cases`
	.page `https://www.adidas.co.uk/`;
test('Click on Carousel item', async t => {
    await t.maximizeWindow()
    .expect(hp.carousel.exists).ok()
    .click(hp.consentCloseButton)
    if(await hp.carouselItem.exists && hp.carouselItem.visible){
    	   await t
    	   .hover(hp.carouselItem)
    	   .click(hp.carouselItem);
    }

});
```
#### pageObject
```
import { Selector } from 'testcafe';

export default class HomePage {

    constructor() {
        this.heroImage = Selector('section[class*="main"] [class*="content"]>[class*="wallpaperWrapper"]');
        this.carousel = Selector('[data-auto-id="carousel-slider"]');
        this.carouselItem = Selector('[data-auto-id="carousel-slider"]>div>div[class*="hockeycard"] div.gl-product-card__media').child('a').nth(0);
        this.consentCloseButton = Selector('#truste-consent-button');
    }

}
```
Tested page URL: https://www.adidas.co.uk/

Test code

### Specify your

* operating system:Browser: Chrome 68.0.3440 / Windows 7.0.0

* testcafe version: 0.22.
* node.js version:8.11

**Comment by AndreyBelym** [comment]
Hi @KristianPP, I've discovered that we have another problem that happens on `https://adidas.co.uk`: https://github.com/DevExpress/testcafe-hammerhead/issues/1769.

We need time to fix it first before reproducing the scrolling issue. Could you please specify another website with the same scrolling behavior?

**Comment by KristianPP** [comment]
Hi @AndreyBelym  thanks for the prompt answer.
http://www.bbc.com/ is the other site but just to note here i see other fails as well.
There were a few other sites i tried but i'll need to check which elements i wanted to use.

The common part of those sites as far as i'm aware is they use reactjs or similar libraries.

Regards,
Kris.

**Comment by AndreyBelym** [comment]
Unfortunately, I can't reproduce the problem on `http://www.bbc.com`. Could you please provide me a complete test file that can be used to reproduce the problem?

**Comment by miherlosev** [comment]
The fix for DevExpress/testcafe-hammerhead#1769 was published in `testcafe@0.23.0`. 
Try to run tests with this TestCafe version. If the problem persists, please, provide us a simple example in witch  we can reproduce the problem.

**Comment by miherlosev** [comment]
I will close the issue because no activity has been encountered for a long time.
